src/app/components/header/header.component.ts

```diff
@@ -7,4 +8,13 @@ import { Component } from '@angular/core';
 })
 export class HeaderComponent {
 
+  constructor(private dataStorageService: DataStorageService) {}
+
+  onSaveData() {
+    this.dataStorageService.storeRecipes();
+  }
+
+  onFetchData() {
```

Comment:
not sure that you need just to fetch data here, probably it will be usefull if you will store this data somewhere (probably, using https://ngrx.io store for whole application), because now you are subscribed, but doing nothing with this data